app/models/manageiq/providers/kubernetes/container_manager/metrics_capture.rb

```
@@ -1,10 +1,60 @@
+require 'hawkularclient'
+
 module ManageIQ::Providers
   class Kubernetes::ContainerManager::MetricsCapture < BaseManager::MetricsCapture
+    class CollectionFailure < RuntimeError; end
+    class TargetValidationError < RuntimeError; end
+
+    require_nested :CaptureContext
+
+    INTERVAL   = 20.seconds
+
+    VIM_STYLE_COUNTERS = {
+      "cpu_usage_rate_average" => {
+        :counter_key           => "cpu_usage_rate_average",
+        :instance              => "",
+        :capture_interval      => "#{INTERVAL}",
+        :precision             => 1,
+        :rollup                => "average",
+        :unit_key              => "percent",
+        :capture_interval_name => "realtime"
+      },
+      "mem_usage_absolute_average" => {
+        :counter_key           => "mem_usage_absolute_average",
+        :instance              => "",
+        :capture_interval      => "#{INTERVAL}",
+        :precision             => 1,
+        :rollup                => "average",
+        :unit_key              => "percent",
+        :capture_interval_name => "realtime"
+      }
+    }
+
     def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
+      start_time ||= 15.minutes.ago.beginning_of_minute.utc
+
       target_name = "#{target.class.name.demodulize}(#{target.id})"
       _log.info("Collecting metrics for #{target_name} [#{interval_name}] " \
                 "[#{start_time}] [#{end_time}]")
-      [{target.ems_ref => {}}, {target.ems_ref => {}}]
+
+      context = CaptureContext.new(target, start_time, end_time, INTERVAL)
+
+      Benchmark.realtime_block(:collect_data) do
+        begin
+          context.collect_metrics
+        rescue CollectionFailure => e
+          _log.error("Hawkular metrics service unavailable: #{e.message}")
```

Should this error out instead of continuing along?